model/task/task.go

```diff
@@ -214,6 +217,16 @@ type Task struct {
 	testResultsPopulated bool
 }
 
+// ExecutionMode indicates the type of environment that the task runs in.
+type ExecutionMode string
+
+const (
+	// ExecutionModeHost indicates that the task runs in a host.
+	ExecutionModeHost ExecutionMode = "host"
+	// ExecutionModeContainer indicates that the task runs in a container.
+	ExecutionModeContainer ExecutionMode = "container"
+)
```

Comment:
Do we foresee adding more execution modes? I think "ExecutionMode" might be a little vague, given that the information in here is basically "does this task run on a container or a host". Maybe ExecutionPlatform?

Reply:
I don't foresee adding more today, but it could change in the future. I find that using booleans (e.g. `IsHostTask`) in data models instead of enums is pretty inflexible to change and I've seen numerous cases where a boolean was used to represent one specific feature and then later seems like a bad decision in hindsight when it morphs into many booleans that you have to check when a single enum would have achieved the same result with less pain. Since we can't do migrations against the task collection easily, future-proofing is the easiest way to avoid making the task model more complicated than it already is.

I don't have a good name for this, so I renamed it to ExecutionPlatform.

Reply:
Agreed that enums seems like the best approach. ExecutionPlatform sounds better to me.